stm32_base/hal/usart_stm32/usart_device_stm32.c

```diff
@@ -729,6 +739,15 @@ void USART2_IRQHandler(void)
 //		xQueueSendFromISR( xUsart2Rx, &cChar, &xTaskWokenByPost );
     }
 
+    if (USART_GetITStatus(USART2, USART_IT_ORE_RX) != RESET) {
```

Comment:
I did this on purpose to make the IRQ handler as fast as possible. inline keyword maybe?